irohad/ordering/impl/ordering_gate_cache/on_demand_cache.cpp

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ordering/impl/ordering_gate_cache/on_demand_cache.hpp"
+
+using namespace iroha::ordering::cache;
+
+void OnDemandCache::addToBack(
+    const OrderingGateCache::BatchesSetType &batches) {
+  std::unique_lock<std::shared_timed_mutex> lock(mutex_);
```

Comment:
I think it is matter of the taste of developer. Also here it is part of the logic since we should tell for the code readers that this method can only be executed synchronously with other methods in that class. Achieving the same result by adding more instances to the code will only make it more complicated. Any strong objectives to refactor that code?

Reply:
Objectives are the following: the code semantically does both synchronization and business logic. From a test perspective better to create tests for bl and tests for synchronization separately.

You can check it out how it is done in other modules. For example, `mst_storage.cpp`.